patches/tModLoader/Terraria/Main.cs.patch

```diff
@@ -6701,6 +6710,57 @@
  		if (gameMenu || player[myPlayer].gravDir == 1f)
  			Rasterizer = RasterizerState.CullCounterClockwise;
  		else
+@@ -51113,7 +_,11 @@
+ 		sceneArea.totalWidth = screenWidth;
+ 		sceneArea.SceneLocalScreenPositionOffset = Vector2.Zero;
+ 		SceneArea sceneArea2 = sceneArea;
++		bool drawStars = true;
++		bool drawSunAndMoon = true;
++		bool drawSky = gameMenu ? MenuLoader.CurrentMenu.PreDrawSky(spriteBatch, ref drawStars, ref drawSunAndMoon) : true;
+ 		if (!mapFullscreen) {
++			if (drawSky) {
+ 			unityMouseOver = false;
```

Comment:
This will get accidentally auto-formatted. Use goto statements to jump over what you have defined as a block.